bpf/aya-bpf/src/lib.rs

```diff
@@ -58,18 +59,17 @@ pub trait BpfContext {
 
 #[no_mangle]
 pub unsafe extern "C" fn memset(s: *mut u8, c: c_int, n: usize) {
-    let base = s as usize;
+    #[allow(clippy::cast_sign_loss)]
+    let b = c as u8;
     for i in 0..n {
-        *((base + i) as *mut u8) = c as u8;
+        *s.add(i) = b;
```

Comment:
I'm not sure we can use add in mem builtins, see the allocated object remark here https://doc.rust-lang.org/stable/std/primitive.pointer.html#safety-7

Reply:
I'm not sure what to make of that remark, what do you think the behavior change would be? Tests seem happy.

Reply:
Where do we go from here? Shall we land and find out?